src/core/js/adapt.js

```diff
@@ -1,431 +1,429 @@
+      if (isViewOnlyQuestion) {
+        // Use question model by default
+        this.log && this.log.deprecated(`Assuming a question model for a view-only question: ${name}`);
+        return 'question';
       }
-      const isBackboneModel = (object.model && object.model.prototype instanceof Backbone.Model);
-      if (!isBackboneModel && object.model instanceof Function) {
-        return object.model();
+      const names = [
+        typeof nameModelOrData._model === 'string' && nameModelOrData._model,
+        typeof nameModelOrData._component === 'string' && nameModelOrData._component,
+        typeof nameModelOrData._type === 'string' && nameModelOrData._type
+      ].filter(Boolean);
+      if (names.length) {
+        // find first fitting model name
+        const name = names.find(name => this.store[name] && this.store[name].model);
+        return name || names.pop(); // return last available if none found
       }
-      return object.model;
     }
+    throw new Error('Cannot derive model class name from input');
+  }
 
-    /**
-     * Looks up which collection a model belongs to
-     * @param {string} id The id of the item you want to look up e.g. `"co-05"`
-     * @return {string} One of the following (or `undefined` if not found):
-     * - "course"
-     * - "contentObjects"
-     * - "blocks"
-     * - "articles"
-     * - "components"
-     */
-    mapById(id) {
-      return this.data.mapById(id);
+  /**
+   * Fetches a model class from the store. For a usage example, see either HotGraphic or Narrative
+   * @param {string|Backbone.Model|object} name The name of the model you want to fetch e.g. `"hotgraphic"` or its json data
+   * @returns {Backbone.Model} Reference to the view class
+   */
+  getModelClass(nameModelOrData) {
+    const name = this.getModelName(nameModelOrData);
+    const object = this.store[name];
+    if (!object) {
+      this.log.warnOnce(`A model for '${name}' isn't registered in your project`);
+      return;
     }
-
-    /**
-     * Looks up a model by its `_id` property
-     * @param {string} id The id of the item e.g. "co-05"
-     * @return {Backbone.Model}
-     */
-    findById(id) {
-      return this.data.findById(id);
+    const isBackboneModel = (object.model && object.model.prototype instanceof Backbone.Model);
+    if (!isBackboneModel && object.model instanceof Function) {
+      return object.model();
     }
+    return object.model;
+  }
 
-    findViewByModelId(id) {
-      const model = this.data.findById(id);
-      if (!model) return;
+  /**
+   * Looks up which collection a model belongs to
+   * @param {string} id The id of the item you want to look up e.g. `"co-05"`
+   * @return {string} One of the following (or `undefined` if not found):
+   * - "course"
+   * - "contentObjects"
+   * - "blocks"
+   * - "articles"
+   * - "components"
+   */
+  mapById(id) {
+    return this.data.mapById(id);
+  }
```

Comment:
suggest either removing this or at least adding a deprecated warning

https://github.com/adaptlearning/adapt_framework/issues/2908

Reply:
Can I do that in a different pr?

Reply:
Have removed.